pkg/cloud/services/ec2/subnets.go

```diff
@@ -146,6 +133,65 @@ LoopExisting:
 	return nil
 }
 
+func (s *Service) getDefaultSubnets() (infrav1.Subnets, error) {
+	zones, err := s.getAvailableZones()
+	if err != nil {
+		return nil, err
+	}
+
+	maxZones := defaultMaxNumAZs
+	if s.scope.VPC().AvailabilityZoneUsageLimit != nil {
+		maxZones = *s.scope.VPC().AvailabilityZoneUsageLimit
+	}
+	selectionScheme := infrav1.AZSelectionSchemeOrdered
+	if s.scope.VPC().AvailabilityZoneSelection != nil {
+		selectionScheme = *s.scope.VPC().AvailabilityZoneSelection
+	}
```

Comment:
Since we are defaulting these values via OpenAPI/kubebuilder annotations, I'm wondering if it would make more sense to error if these are not set here rather than potentially having to worry about the openapi defaults and the defaults encoded here diverging

Reply:
I think anything that guards against divergence would be good